### ISSUE TITLE
InlineDerivedTable can't be outer joined

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15629]
I've added these two integration tests where an `InlineDerivedTable` is used in an outer join:

```java
public void testInlineDerivedTableLeftJoined() {
    TBook b = T_BOOK.where(T_BOOK.ID.in(1, 2));

    Result<Record2<Integer, Integer>> r =
    create()
        .select(T_LANGUAGE.ID, b.ID)
        .from(T_LANGUAGE)
        .leftJoin(b).on(b.LANGUAGE_ID.eq(T_LANGUAGE.ID))
        .orderBy(T_LANGUAGE.ID, b.ID)
        .fetch();

    assertEquals(asList(1, 1, 2, 3, 4), r.map(x -> x.value1()));
    assertEquals(asList(1, 2, null, null, null), r.map(x -> x.value2()));
}

public void testInlineDerivedTableRightJoined() {
    assumeSupports(() -> T_LANGUAGE.rightJoin(T_BOOK));

    TBook b = T_BOOK.where(T_BOOK.ID.in(1, 2));

    Result<Record2<Integer, Integer>> r =
    create()
        .select(T_LANGUAGE.ID, b.ID)
        .from(b)
        .rightJoin(T_LANGUAGE).on(b.LANGUAGE_ID.eq(T_LANGUAGE.ID))
        .orderBy(T_LANGUAGE.ID, b.ID)
        .fetch();

    assertEquals(asList(1, 1, 2, 3, 4), r.map(x -> x.value1()));
    assertEquals(asList(1, 2, null, null, null), r.map(x -> x.value2()));
}
```

Both queries fail with a syntax error, e.g.

```sql
select PUBLIC.T_LANGUAGE.ID, PUBLIC.T_BOOK.ID
from PUBLIC.T_LANGUAGE
  left outer join (
    select *
    from PUBLIC.T_BOOK
    where PUBLIC.T_BOOK.ID = 1
  ) T_BOOK
    on PUBLIC.T_BOOK.LANGUAGE_ID = PUBLIC.T_LANGUAGE.ID
order by PUBLIC.T_LANGUAGE.ID, PUBLIC.T_BOOK.ID
```

Raises this error on H2:

> Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "PUBLIC.T_BOOK.ID" not found; SQL statement:

The problem is that we have to create a derived table to maintain outer join semantics, and that derived table is named `T_BOOK`, not `PUBLIC.T_BOOK`.

Possible solutions:

- Stop qualifying references to all `InlineDerivedTable` column references. This would be a simple fix (probably), but risks creating ambiguities when two tables from different schemas share the same name (which is probably rare but not impossible).
- Always generate an alias for inline derived tables, similar to implicit join paths. This wouldn't create ambiguities, but would require users to dereference columns from the inline derived table instead of the base table, which they might not be doing currently.

I don't see any other solutions, and among the two, the first one is more compelling, because less risky of regressions. But still, given the slight risk, I won't backport the fix.

The main driver for this improvement is the fact that the new jOOQ 3.19 `Policy` feature will be implemented using `InlineDerivedTable`:

- https://github.com/jOOQ/jOOQ/issues/2682

**Comment by lukaseder** [comment]
Users can still request a backport later if they have a compelling use-case. Though, I doubt this feature is used a lot in user code.

Fixed for jOOQ 3.19

**Comment by lukaseder** [comment]
A related problem was solved in https://github.com/jOOQ/jOOQ/issues/8012, where `Table::fields` got an alternative implementation in order to qualify fields correctly.
